### PR TITLE
Make sure that --generate-rcfile generates a valid file

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Ref. https://github.com/PyCQA/pylint/pull/6529

Fixes the issues with `ini`.

### CODE REVIEW

File: tests/test_self.py [hunk 1]
```diff
@@ -1344,6 +1345,14 @@ def test_generate_rcfile() -> None:
         )
         assert process.stdout == process_two.stdout
 
+        with tempfile.NamedTemporaryFile(mode="w") as temp:
```
Comment (DanielNoord) [inline tests/test_self.py:1348]:
```suggestion
        # Check that the generated file is valid
        with tempfile.NamedTemporaryFile(mode="w") as temp:`
```

File: tests/test_self.py [hunk 2]
```diff
@@ -1344,6 +1345,15 @@ def test_generate_rcfile() -> None:
         )
         assert process.stdout == process_two.stdout
 
+        # Check that the generated file is valid
+        with tempfile.NamedTemporaryFile(mode="w") as temp:`
```
Comment (DanielNoord) [inline tests/test_self.py:1349]:
```suggestion
        with tempfile.NamedTemporaryFile(mode="w") as temp:
```

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas This passes tests and should fix the last issues in the original PR.

**Review by Pierre-Sassoulas — APPROVED** [review]
👍 Let's wrap up 2.14-beta :)

**Pull request merged**